Adds additional options for ingest requests

cwingard opened the pull request:
Adds stage and mock as ingest request types and adds ability to specify a starting and/or ending date and time to bound the ingests (based on instrument time, not file modification time).

cwingard committed:
Adds additional options for ingest requests

Adds stage and mock as ingest request types and adds ability to specify a starting and/or ending date and time to bound the ingests (based on instrument time, not file modification time).

cdobs commented on ingestions/ingest_csvs.py:
```diff
@@ -136,15 +138,20 @@ def main(argv=None):
     parser.add_argument("-c", "--csvfile", dest="csvfile", type=Path, required=True)
-    parser.add_argument("-t", "--ingest_type", dest="ingest_type", type=str, choices=('recovered', 'telemetered'),
+    parser.add_argument("-t", "--ingest_type", dest="ingest_type", type=str, 
+                        choices=('recovered', 'telemetered', 'stage', 'mock'), 
```
Stage and mock should be options for "state" within the ingest request. Options for type are recovered, telemetered, and playback. Options for state are stage, mock, run.

cwingard replied:
OK, these have been properly split out, with run set as the default state.

cdobs reviewed (changes requested):
Changes to beginData/endData look good but the code should be updated to add mock and stage as options for "state" within the ingest request and not "type"

cwingard committed:
Updates arguments to add ingest state as an option

Allows the user to specify the ingest state as either run, stage or mock. The default state is set to run. Additionally, adds help strings to the arguments.

cdobs reviewed (approved):
All changes look good. Tested script with new flags and it works great. Thanks Chris!
